### DISCUSSION TITLE
feat: list user notifications/status

### DISCUSSION THREAD

**Discussion by tmillr** [pwntester/octo.nvim#387] [Ideas]
There's no command to list your GitHub notifications?

E.g. something similar to `gh status` and/or a command to list your user notifications (and views them, either in browser or buffer, on `<CR>`) would be nice.

**Comment by pwntester** [comment]
Sorry for the late response, that has not been implemented yet. Unfortunately GitHub does not expose an API to mark notifications as Read/Done yet

**Reply by tmillr** [reply]
Ah ok. It would still be cool to be able to view them and follow them when clicked (even if only in the browser). Like for quickly checking for new notifications. But maybe this would be better implemented as a telescope plugin, or system notification driver.

**Comment by williambdean** [comment]
Hi, @tmillr

Notifications are available via `Octo notification list` command. Status is also something that is easy to implement via the `commands` section of the configuration file. I do that for my own config here: https://github.com/wd60622/dotfiles/blob/c2f239bed4a8ad646e06c2c2537f06c628530cf1/nvim/lua/plugins/version-control.lua?plain=1#L289-L295

**Reply by tmillr** [reply]
Thanks for the response, I'll look into `Octo notification list`. Does that command list all of your [user notifications](https://github.com/notifications), or is it only notifications from the current git repo?

**Reply by williambdean** [reply]
That depends on the configuration. There is a "current_repo_only" configuration option. 

You can also have full control by using `require "octo.picker".notifications {}` directly

**Reply by tmillr** [reply]
Nice find. Though it doesn't seem to match the web UI for some reason (I think some of my unread notifications are missing from it). Maybe gh doesn't offer this exact interface, or maybe it only exists through the REST api?

I'll dig deeper when I get the chance, and check out `require "octo.picker".notifications`, and the gh API docs again.

**Reply by williambdean** [reply]
There is also a flag "all=true"
